### DISCUSSION TITLE
Pressure of liquid on elastic container surface

### DISCUSSION THREAD

**Discussion by ChangjingLiu** [sofa-framework/sofa#3262] [Write / Run a simulation] [answered]
I want to simulate a scene that injecting a certain amount of liquid into the elastic container, and I could get pressure from different depth of the surface of the container.

I've known a component called `TaitSurfacePressureForceField`, but it is suitble for the above scene?

**Comment by damienmarchal** [comment]
I don't know what do TaitSurfacePressureForceField but this remind me of a component made few years ago by @EulalieCoevoet. I don't know if this one landed into a public plugin or didn't (yet).

**Reply by EulalieCoevoet** [reply, marked as answer]
Indeed @damienmarchal we made a component that could match your need @Changjing-Liu. It's the implementation of [this paper](https://hal.inria.fr/hal-01535810/document), so with the hypothesis is that the cavity is always full of water, we do not model the fluid, just the force it applies on the wall of the container. However, this code has not been maintained unfortunately, so it is not usable I'm sorry.
